### ISSUE TITLE
Difficulty starting rethinkdb-proxy-28015.service

### ISSUE THREAD

**Issue by MichaelMadsen** [openstf/stf#73]
Hi there
Again, hope I'm not missing something obvious, but it turned out to be a good idea to write here when I couldn't get adbd.service working.

When I start rethinkdb-proxy-28015 I get an error that I have tried to troubleshoot all day.
Using the same rethinkdb-proxy-28015.service as DEPLOYMENT.md, changed the authkey to 1234 and the ip to the ip of this same machine (running everything on one machine)
RethinkDB is running with --bind all, and I can access the web interface from my browser on the same ip, port 8080.

Here is what happens when I try to start the service:
  ```
Sep 07 15:44:26 ubuntu systemd[1]: Starting rethinkdb-proxy-28015.service...
Sep 07 15:44:28 ubuntu docker[568]: latest: Pulling from ctlc/ambassador
Sep 07 15:44:28 ubuntu docker[568]: 093e01545ca5: Already exists
Sep 07 15:44:28 ubuntu docker[568]: 639d60300768: Already exists
Sep 07 15:44:28 ubuntu docker[568]: 50e9f95f98f1: Already exists
Sep 07 15:44:28 ubuntu docker[568]: 6d021018145f: Already exists
Sep 07 15:44:28 ubuntu docker[568]: c33211982f7d: Already exists
Sep 07 15:44:28 ubuntu docker[568]: 0da04ef149d0: Already exists
Sep 07 15:44:28 ubuntu docker[568]: c56de9c5a0d2: Already exists
Sep 07 15:44:28 ubuntu docker[568]: 410000273ccc: Already exists
Sep 07 15:44:28 ubuntu docker[568]: 1c9abde34886: Already exists
Sep 07 15:44:28 ubuntu docker[568]: 53e2694bc9d6: Already exists
Sep 07 15:44:28 ubuntu docker[568]: a853fbf35ee2: Already exists
Sep 07 15:44:28 ubuntu docker[568]: cea471227545: Already exists
Sep 07 15:44:28 ubuntu docker[568]: cea471227545: Already exists
Sep 07 15:44:28 ubuntu docker[568]: Digest: sha256:f1648fa46b91d757b7cdd681ff134284e9903849faab0d30f3d8179c2429340c
Sep 07 15:44:28 ubuntu docker[568]: Status: Image is up to date for ctlc/ambassador:latest
Sep 07 15:44:28 ubuntu docker[701]: Error response from daemon: Cannot kill container rethinkdb-proxy-28015: no such id: rethinkdb-proxy-28015
Sep 07 15:44:28 ubuntu docker[701]: Error: failed to kill containers: [rethinkdb-proxy-28015]
Sep 07 15:44:28 ubuntu docker[709]: Error response from daemon: no such id: rethinkdb-proxy-28015
Sep 07 15:44:28 ubuntu docker[709]: Error: failed to remove containers: [rethinkdb-proxy-28015]
Sep 07 15:44:28 ubuntu systemd[1]: Started rethinkdb-proxy-28015.service.
Sep 07 15:44:28 ubuntu docker[716]: 2015-09-07 14:44:28,963 CRIT Supervisor running as root (no user in config file)
Sep 07 15:44:28 ubuntu docker[716]: 2015-09-07 14:44:28,963 WARN Included extra file "/etc/supervisor/conf.d/supervisord-socat.conf" during parsing
Sep 07 15:44:28 ubuntu docker[716]: 2015-09-07 14:44:28,988 INFO RPC interface 'supervisor' initialized
Sep 07 15:44:28 ubuntu docker[716]: 2015-09-07 14:44:28,988 WARN cElementTree not installed, using slower XML parser for XML-RPC
Sep 07 15:44:28 ubuntu docker[716]: 2015-09-07 14:44:28,988 CRIT Server 'unix_http_server' running without any HTTP authentication checking
Sep 07 15:44:28 ubuntu docker[716]: 2015-09-07 14:44:28,988 INFO supervisord started with pid 1
Sep 07 15:44:29 ubuntu docker[716]: 2015-09-07 14:44:29,992 INFO spawned: 'socat' with pid 7
Sep 07 15:44:30 ubuntu docker[716]: 2015-09-07 14:44:29,999 INFO exited: socat (exit status 0; not expected)
Sep 07 15:44:31 ubuntu docker[716]: 2015-09-07 14:44:31,002 INFO spawned: 'socat' with pid 13
Sep 07 15:44:31 ubuntu docker[716]: 2015-09-07 14:44:31,009 INFO exited: socat (exit status 0; not expected)
Sep 07 15:44:31 ubuntu docker[716]: 2015-09-07 14:44:31,012 CRIT reaped unknown pid 18)
Sep 07 15:44:33 ubuntu docker[716]: 2015-09-07 14:44:33,018 INFO spawned: 'socat' with pid 22
Sep 07 15:44:33 ubuntu docker[716]: 2015-09-07 14:44:33,026 INFO exited: socat (exit status 0; not expected)
Sep 07 15:44:33 ubuntu docker[716]: 2015-09-07 14:44:33,027 CRIT reaped unknown pid 27)
Sep 07 15:44:36 ubuntu docker[716]: 2015-09-07 14:44:36,032 INFO spawned: 'socat' with pid 28
Sep 07 15:44:36 ubuntu docker[716]: 2015-09-07 14:44:36,046 INFO exited: socat (exit status 0; not expected)
Sep 07 15:44:36 ubuntu docker[716]: 2015-09-07 14:44:36,048 INFO gave up: socat entered FATAL state, too many start retries too quickly
Sep 07 15:44:36 ubuntu docker[716]: 2015-09-07 14:44:36,048 CRIT reaped unknown pid 33)
```

And here is my rethinkdb-proxy-28015.service file:
```
Description=RethinkDB proxy/28015
After=docker.service
Requires=docker.service

[Service]
EnvironmentFile=/etc/environment
TimeoutStartSec=0
Restart=always
ExecStartPre=/usr/bin/docker pull ctlc/ambassador:latest
ExecStartPre=-/usr/bin/docker kill %p
ExecStartPre=-/usr/bin/docker rm %p
ExecStart=/usr/bin/docker run --rm \
  --name %p \
  -e "AUTHKEY=1234" \
  -p 28015 \
  -e RETHINKDB_PORT_28015_TCP=tcp://192.168.99.106:28015 \
  ctlc/ambassador:latest
ExecStop=-/usr/bin/docker stop -t 10 %p
```

Tried running the docker run command directly, both on this machine and on a fresh install and it does the same. I am unsure where I should look to find the error, feel like I've been through everything.

Thanks for your time, I appreciate the help

**Comment by bk10aao** [comment]
Hi,

I am experiencing the same problem. 

Can anyone help?

Thanks!

**Comment by bk10aao** [comment]
-- Logs begin at Mon 2015-09-07 22:18:29 BST, end at Tue 2015-09-08 15:56:09 BST. --
Sep 08 15:53:32 ubuntu systemd[1]: Starting RethinkDB proxy/28015...
Sep 08 15:53:34 ubuntu docker[2464]: latest: Pulling from ctlc/ambassador
Sep 08 15:53:34 ubuntu docker[2464]: 093e01545ca5: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: 639d60300768: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: 50e9f95f98f1: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: 6d021018145f: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: c33211982f7d: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: 0da04ef149d0: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: c56de9c5a0d2: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: 410000273ccc: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: 1c9abde34886: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: 53e2694bc9d6: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: a853fbf35ee2: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: cea471227545: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: cea471227545: Already exists
Sep 08 15:53:34 ubuntu docker[2464]: Digest: sha256:f1648fa46b91d757b7cdd681ff134284e9903849faab0d30f3d8179c2429340c
Sep 08 15:53:34 ubuntu docker[2464]: Status: Image is up to date for ctlc/ambassador:latest
Sep 08 15:53:34 ubuntu docker[2476]: Error response from daemon: Cannot kill container rethinkdb-proxy-28015: Container c914ebf62e08487ab381cd32330c0eaae730dd5e25d66c7477f06688b0fbc
Sep 08 15:53:34 ubuntu docker[2476]: Error: failed to kill containers: [rethinkdb-proxy-28015]
Sep 08 15:53:34 ubuntu docker[2483]: rethinkdb-proxy-28015
Sep 08 15:53:34 ubuntu systemd[1]: Started RethinkDB proxy/28015.
Sep 08 15:53:35 ubuntu docker[2490]: 2015-09-08 14:53:35,291 CRIT Supervisor running as root (no user in config file)
Sep 08 15:53:35 ubuntu docker[2490]: 2015-09-08 14:53:35,291 WARN Included extra file "/etc/supervisor/conf.d/supervisord-socat.conf" during parsing
Sep 08 15:53:35 ubuntu docker[2490]: 2015-09-08 14:53:35,327 INFO RPC interface 'supervisor' initialized
Sep 08 15:53:35 ubuntu docker[2490]: 2015-09-08 14:53:35,328 WARN cElementTree not installed, using slower XML parser for XML-RPC
Sep 08 15:53:35 ubuntu docker[2490]: 2015-09-08 14:53:35,328 CRIT Server 'unix_http_server' running without any HTTP authentication checking
Sep 08 15:53:35 ubuntu docker[2490]: 2015-09-08 14:53:35,329 INFO supervisord started with pid 1
Sep 08 15:53:36 ubuntu docker[2490]: 2015-09-08 14:53:36,332 INFO spawned: 'socat' with pid 6
Sep 08 15:53:36 ubuntu docker[2490]: 2015-09-08 14:53:36,353 INFO exited: socat (exit status 0; not expected)
Sep 08 15:53:37 ubuntu docker[2490]: 2015-09-08 14:53:37,357 INFO spawned: 'socat' with pid 12
Sep 08 15:53:37 ubuntu docker[2490]: 2015-09-08 14:53:37,367 INFO exited: socat (exit status 0; not expected)
Sep 08 15:53:37 ubuntu docker[2490]: 2015-09-08 14:53:37,370 CRIT reaped unknown pid 17)
Sep 08 15:53:39 ubuntu docker[2490]: 2015-09-08 14:53:39,375 INFO spawned: 'socat' with pid 18
Sep 08 15:53:39 ubuntu docker[2490]: 2015-09-08 14:53:39,383 INFO exited: socat (exit status 0; not expected)
Sep 08 15:53:39 ubuntu docker[2490]: 2015-09-08 14:53:39,386 CRIT reaped unknown pid 23)
Sep 08 15:53:42 ubuntu docker[2490]: 2015-09-08 14:53:42,391 INFO spawned: 'socat' with pid 24
Sep 08 15:53:42 ubuntu docker[2490]: 2015-09-08 14:53:42,404 INFO exited: socat (exit status 0; not expected)
Sep 08 15:53:42 ubuntu docker[2490]: 2015-09-08 14:53:42,406 INFO gave up: socat entered FATAL state, too many start retries too quickly
Sep 08 15:53:42 ubuntu docker[2490]: 2015-09-08 14:53:42,406 CRIT reaped unknown pid 29)
Sep 08 15:53:51 ubuntu systemd[1]: Started RethinkDB proxy/28015.
Sep 08 15:56:09 ubuntu systemd[1]: Started RethinkDB proxy/28015.

**Comment by sorccu** [comment]
Looks like there was a commit last month and the ambassador image got rebuilt. Something probably changed and broke things.

* https://github.com/CenturyLinkLabs/ctlc-docker-ambassador
* https://hub.docker.com/r/ctlc/ambassador/

**Comment by benjamin-m-kane** [comment]
Thanks for the reply,

I have pulled the previous version and the only difference between that and the latest is the addition of a line:

https://github.com/benjamin-m-kane/ctlc-docker-ambassador/commit/a7e01769b4c35859f8ebf606c3fa452fc8e88d9f

Any further guidance/advice would be appreciated.

Thanks.

**Comment by sorccu** [comment]
I know that only one line changed, but it caused a full rebuild of the docker image. Something must have changed during that rebuild.

**Comment by vbanthia** [comment]
Well just to add, I just deployed STF today with latest ambassador image and it worked for me.

**Comment by sorccu** [comment]
I'm trying it at home and it's not working. But I know for certain that we've been using the same unit (with automatic docker pull) for probably around 2 years and it has always worked, even on new machines. But it looks like we'll have to find an alternative now.

**Comment by sorccu** [comment]
Oh... you know guys, it actually works. The image is buggy and you do get the errors every time. However, it still manages to open the link behind the scenes. Anyway that's bad, so I'll try think of something.

**Comment by sorccu** [comment]
Please try to change your unit configurations to use `openstf/ambassador`. Other than that the usage stays the same.

**Comment by maykel535** [comment]
Impossible launche stf-app.service with rethinkdb-proxy-28015 .... When launch rethinkdb and rethinkdb-proxy-28015 works fine, but when launche stf-app.service the load average and cpu is high. socat use high cpu...

# systemctl status rethinkdb-proxy-28015.service

feb 22 19:01:22 nodito1 systemd[1]: Started RethinkDB proxy/28015.
feb 22 19:01:32 nodito1 docker[30460]: 2018/02/22 18:01:32 socat[28244] E connect(5, AF=2 172.17.0.2:28015, 16): Address not available
feb 22 19:01:55 nodito1 docker[30460]: 2018/02/22 18:01:55 socat[24009] E connect(5, AF=2 172.17.0.2:28015, 16): Address not available

From host I can telnet to ip and port works well:

@nodito1:/lib/systemd/system# telnet 172.17.0.2 28015
Trying 172.17.0.2...
Connected to 172.17.0.2.
Escape character is '^]'.

I don't understanding anything...

[https://imgur.com/a/54Niw](url)

**Comment by maykel535** [comment]
I solved change the ip 172.17.0.2 from docker to ip from host physical and works well...

I don't understand ...but works now
